go/trace/logger_test.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trace
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLoggerLogAndError(t *testing.T) {
+	logger := traceLogger{}
+
+	// Redirect stderr to a buffer
+	rescueStderr := os.Stderr
+	r, w, _ := os.Pipe()
+	os.Stderr = w
+
+	want := "This is an error message"
+	logger.Error(want)
+
+	w.Close()
+	got, _ := io.ReadAll(r)
+	os.Stderr = rescueStderr
+
+	assert.Contains(t, string(got), want)
+
+	r, w, _ = os.Pipe()
+	os.Stderr = w
+
+	want = "This is an log message"
+	logger.Log(want)
+
+	w.Close()
+	got, _ = io.ReadAll(r)
+	os.Stderr = rescueStderr
+
+	assert.Contains(t, string(got), want)
```

Comment:
Instead of having such a long function, it is better to make test cases, that will make the setup code a lot smaller, since it will be shared. Also the test is far more readable this way, since the expecatation and the parameters are listed together.

Reply:
Done. Thanks for the advice.